common/sketch/src/main/java/org/apache/spark/util/sketch/BitArray.java

```diff
@@ -32,13 +38,14 @@ static int numWords(long numBits) {
   }
 
   BitArray(long numBits) {
-    if (numBits <= 0) {
-      throw new IllegalArgumentException("numBits must be positive");
-    }
-    this.data = new long[numWords(numBits)];
+    this(new long[numWords(numBits)]);
+  }
+
+  private BitArray(long[] data) {
+    this.data = data;
     long bitCount = 0;
-    for (long value : data) {
-      bitCount += Long.bitCount(value);
+    for (long datum : data) {
```

Comment:
it is a little bit weird to say datam here, since you are actually working with 64 "datum" at once. maybe "word"?